core/src/main/scala/dagr/core/tasksystem/Pipeline.scala

```diff
@@ -78,9 +80,15 @@ abstract class Pipeline(val outputDirectory: Option[Path] = None,
 
   /** Recursively navigates dependencies, starting from the supplied task, and add all children to this.tasks. */
   private def addChildren(task : Task) : Unit = {
-    task.tasksDependingOnThisTask.filterNot(tasks.contains).foreach { child =>
-      tasks += child
-      addChildren(child)
+    // 1. find all tasks connected to this task
+    val toVisit: mutable.Set[Task] = mutable.HashSet[Task](task)
```

Comment:
Why use a `Set` here instead of a `Stack`?  Are you gaining something from the uniqueness checking?  Otherwise it's just overhead.

Reply:
We are gaining from uniqueness checking (I'll add a comment).  Suppose we have `A ==> (B :: C)` and `B ==> C`.  Even thought this could be simplified to `A ==> B ==> C`, that's up to the caller, and we post-processing of the DAG.  So when `addChildren` gets called on `A`, it recurses on `B` and `C`.  Since `C` depends on `C`, without the uniqueness check we recurse on `C` in the `addChildren` call on `B`.

Reply:
Got it, thanks!